app/src/main/java/fr/free/nrw/commons/nearby/NearbyActivity.java

```diff
@@ -42,28 +44,46 @@
 
     @BindView(R.id.progressBar)
     ProgressBar progressBar;
-    private boolean isMapViewActive = false;
     private static final int LOCATION_REQUEST = 1;
+    public static final String MAP_LAST_USED_PREFERENCE = "mapLastUsed";
 
     private LocationServiceManager locationManager;
     private LatLng curLatLang;
     private Bundle bundle;
     private NearbyAsyncTask nearbyAsyncTask;
+    private SharedPreferences sharedPreferences;
+    private NearbyActivityMode viewMode;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
         setContentView(R.layout.activity_nearby);
         ButterKnife.bind(this);
         checkLocationPermission();
         bundle = new Bundle();
         initDrawer();
+        initViewMode();
+    }
+
+    private void initViewMode() {
```

Comment:
If you were to store this preference as a String, you could use the built-in "valueOf()" method of enum.  Alternatively, this method should probably move to the NearbyActivityMode class.  In either case initializing the viewMode could be one line of code.